src/main/java/org/dependencytrack/persistence/ComponentQueryManager.java

```diff
@@ -394,16 +394,20 @@ public void recursivelyDelete(Component component, boolean commitIndex) {
             }
         }
         pm.getFetchPlan().setDetachmentOptions(FetchPlan.DETACH_LOAD_FIELDS);
-        final Component result = pm.getObjectById(Component.class, component.getId());
-        Event.dispatch(new IndexEvent(IndexEvent.Action.DELETE, pm.detachCopy(result)));
-
-        deleteAnalysisTrail(component);
-        deleteViolationAnalysisTrail(component);
-        deleteMetrics(component);
-        deleteFindingAttributions(component);
-        deletePolicyViolations(component);
-        delete(component);
-        commitSearchIndex(commitIndex, Component.class);
+        try {
+            final Component result = pm.getObjectById(Component.class, component.getId());
+            Event.dispatch(new IndexEvent(IndexEvent.Action.DELETE, pm.detachCopy(result)));
+            deleteAnalysisTrail(component);
+            deleteViolationAnalysisTrail(component);
+            deleteMetrics(component);
+            deleteFindingAttributions(component);
+            deletePolicyViolations(component);
+            delete(component);
+            commitSearchIndex(commitIndex, Component.class);
+        } catch (javax.jdo.JDOObjectNotFoundException | org.datanucleus.exceptions.NucleusObjectNotFoundException ignored) {
+
```

Comment:
These exceptions should still be logged, potentially with level `WARNING`. It should be obvious that there's still something going wrong unintentionally.